common/persistence/visibility/store/elasticsearch/query_interceptors.go

```diff
@@ -146,6 +178,44 @@ func (vi *valuesInterceptor) Values(name string, values ...interface{}) ([]inter
 		default:
 		}
 
+		switch fieldType {
```

Comment:
Is this during indexing path or during query path?
We need to document this behavior of what format is supported and what is not. This could potentially be a breaking change if customer rely on unexpected conversion behavior. Do we need to have a dynamic config to disable this enforcement just in case?

Reply:
This is only the in the query path

Reply:
👍 then I'm less concerned. What happen if index path has this invalid values?

Reply:
It already has validation in the index path.